src/builder_state.rs

```diff
@@ -541,48 +542,24 @@ impl<TYPES: BuilderType> BuilderProgress<TYPES> for BuilderState<TYPES> {
             // get the number of quorum committee members to be used for VID calculation
             //let quo_membership = Arc::into_inner(self.quorum_membership).unwrap();//.clone();
 
-            let num_quorum_committee = self.quorum_membership.total_nodes();
-
-            // TODO <https://github.com/EspressoSystems/HotShot/issues/1686>
-            let srs = test_srs(num_quorum_committee);
+            // get the total nodes from the builder state.
+            // stored while processing the DA Proposal
+            let vid_num_nodes = self.total_nodes.get();
 
-            // calculate the last power of two
-            // TODO change after https://github.com/EspressoSystems/jellyfish/issues/339
-            // issue: https://github.com/EspressoSystems/HotShot/issues/2152
-            let chunk_size = 1 << num_quorum_committee.ilog2();
+            // convert vid_num_nodes to usize
+            // spawn a task to calculate the VID commitment, and pass the builder handle to the global state
+            // later global state can await on it before replying to the proposer
+            let join_handle: JoinHandle<<VidScheme as VidSchemeTrait>::Commit> =
```

Comment:
This shouldn't need to be explicitly typed...

Reply:
Done